### DISCUSSION TITLE
Twitch Bot for controlling avatar thru OSC

### DISCUSSION THREAD

**Discussion by Killers0992** [vrchat-community/osc#77] [Show and tell]
Implemented stuff:
- Commands
- OnSubscriber ( its seperate for new subscribers or resubs )
- OnFollow
- OnSendBits
- OnReward ( on reward something thru channel points )

Contains:
- Delay per user or globally
- Some stuff can only executed if viewer is subscribed for x months or have mod/vip rank.
- OSC Actions have option to execute for x amount of seconds and with that every action is queued soo its no way to override current playing OSC action.

Link:
https://github.com/Killers0992/TwitchVrcAvatarOSC

**Comment by momo-the-monster** [comment]
Thanks for sharing this, @Killers0992! It's been added to the [list](https://github.com/vrchat-community/osc/wiki/Resources#twitch).

**Comment by Killers0992** [comment]
@momo-the-monster if this can be removed because that was kinda replaced by  [Link](https://github.com/Killers0992/VrcOscIntegrations) 
same stuff but more interface added for everything and Twitch Integration itself.

**Reply by Killers0992** [reply]
@momo-the-monster
